examples/edge-stack-iso/custom/files/metal3.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+set -euo pipefail
+
+BASEDIR="$(dirname "$0")"
+source ${BASEDIR}/basic-setup.sh
+
+METAL3LOCKNAMESPACE="default"
+METAL3LOCKCMNAME="metal3-lock"
+
+# Get or create the lock to run all those steps just in a single node
+# As the first node is created WAY before the others, this should be enough
+# TODO: Investigate if leases is better
+if [ $(${KUBECTL} get cm -n ${METAL3LOCKNAMESPACE} ${METAL3LOCKCMNAME} -o name | wc -l) -lt 1 ]; then
+  ${KUBECTL} create configmap ${METAL3LOCKCMNAME} -n ${METAL3LOCKNAMESPACE} --from-literal foo=bar
+else
+  exit 0
+fi
+
+# Wait for metal3
+while ! ${KUBECTL} wait --for condition=ready -n ${METAL3_CHART_TARGETNAMESPACE} $(${KUBECTL} get pods -n ${METAL3_CHART_TARGETNAMESPACE} -l app.kubernetes.io/name=metal3-ironic -o name) --timeout=10s; do sleep 2 ; done
+
+# Get the ironic IP
+IRONICIP=$(${KUBECTL} get cm -n ${METAL3_CHART_TARGETNAMESPACE} ironic-bmo -o jsonpath='{.data.IRONIC_IP}')
+
+# This only works for metallb, we need to do the same for nodeport I guess
+
+# Wait for metallb
+while ! ${KUBECTL} wait --for condition=ready -n ${METALLBNAMESPACE} $(${KUBECTL} get pods -n ${METALLBNAMESPACE} -l app.kubernetes.io/component=controller -o name) --timeout=10s; do sleep 2 ; done
+
+# Don't create the ippool if already created
+${KUBECTL} get ipaddresspool -n ${METALLBNAMESPACE} ironic-ip-pool -o name || cat <<-EOF | ${KUBECTL} apply -f -
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: ironic-ip-pool
+  namespace: ${METALLBNAMESPACE}
+spec:
+  addresses:
+  - ${IRONICIP}/32
+  serviceAllocation:
+    priority: 100
+    serviceSelectors:
+    - matchExpressions:
+      - {key: app.kubernetes.io/name, operator: In, values: [metal3-ironic]}
+EOF
+
+# Same for L2 Advs
+${KUBECTL} get L2Advertisement -n ${METALLBNAMESPACE} ironic-ip-pool-l2-adv -o name || cat <<-EOF | ${KUBECTL} apply -f -
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: ironic-ip-pool-l2-adv
+  namespace: ${METALLBNAMESPACE}
+spec:
+  ipAddressPools:
+  - ironic-ip-pool
+EOF
+
+# If clusterctl is not installed, install it
+if ! command -v clusterctl > /dev/null 2>&1; then
+  LINUXARCH=$(uname -m)
+  case $(uname -m) in
+    "x86_64")
+      export GOARCH="amd64" ;;
+    "aarch64")
+      export GOARCH="arm64" ;;
+    "*")
+      echo "Arch not found, asumming amd64"
+      export GOARCH="amd64" ;;
+  esac
+
+  # Clusterctl bin
+  # Maybe just use the binary from hauler if available
```

Comment:
I'd suggest dropping this comment. If we want to sideload the binary, we should instead just put it under `/custom/files`.

Reply:
As hauler allows you to pack binaries (https://rancherfederal.github.io/hauler-docs/docs/guides-references/hauler-content/files) as well I think it would be cleaner if we can add a flow to EIB to save binaries to hauler somehow.

```
hauler store add file https://github.com/kubernetes-sigs/cluster-api/releases/download/v${METAL3_CLUSTERCTLVERSION}/clusterctl-linux-${GOARCH}
```

I can put the binary directly I agree but I don't want to do save it directly into git and if I don't save it the example will not be 'complete'.

Reply:
I see. Keeping the download is completely fine for now. On a separate note, Hauler is intentionally not being used for files right now and this might very well be the case in the future as well but it's something we'll discuss when we have the time.